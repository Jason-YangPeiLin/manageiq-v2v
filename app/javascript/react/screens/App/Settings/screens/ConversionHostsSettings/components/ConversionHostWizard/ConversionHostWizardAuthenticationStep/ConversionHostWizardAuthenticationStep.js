import React from 'react';
import { Form } from 'patternfly-react';

const ConversionHostWizardAuthenticationStep = () => (
  <Form className="form-horizontal">
    <h2>TODO: Authentication Step Contents</h2>
  </Form>
);

export default ConversionHostWizardAuthenticationStep;
