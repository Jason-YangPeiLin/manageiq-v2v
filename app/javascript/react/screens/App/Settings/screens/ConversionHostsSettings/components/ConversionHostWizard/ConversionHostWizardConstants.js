export const stepIDs = {
  locationStep: 'conversionHostWizardLocationStep',
  hostsStep: 'conversionHostWizardHostsStep',
  authenticationStep: 'conversionHostWizardAuthenticationStep',
  resultsStep: 'conversionHostWizardResultsStep'
};
