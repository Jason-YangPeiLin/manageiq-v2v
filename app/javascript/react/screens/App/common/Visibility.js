import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Renders the children wrapped in visibility: hidden if the condition is met.
// Either way, the children take up space, visibility will not change the flow of the layout.
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const Visibility = ({ children, hidden }) => (
  <span style={{ visibility: hidden ? 'hidden' : '' }} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

Visibility.propTypes = {
  children: PropTypes.node,
  hidden: PropTypes.bool
};

export default Visibility;
