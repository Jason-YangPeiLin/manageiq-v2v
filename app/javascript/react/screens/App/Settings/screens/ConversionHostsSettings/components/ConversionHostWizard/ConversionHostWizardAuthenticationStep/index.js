import ConversionHostWizardAuthenticationStep from './ConversionHostWizardAuthenticationStep';

export default ConversionHostWizardAuthenticationStep;
