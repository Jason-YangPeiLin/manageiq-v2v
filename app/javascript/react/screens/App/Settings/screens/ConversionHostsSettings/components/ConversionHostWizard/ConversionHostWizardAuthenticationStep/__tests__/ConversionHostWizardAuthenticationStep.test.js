import React from 'react';
import { shallow, mount } from 'enzyme';
import { reducer as formReducer } from 'redux-form';

import { generateStore } from '../../../../../../../common/testReduxHelpers';
import ConversionHostWizardAuthenticationStep from '../ConversionHostWizardAuthenticationStep';
import { RHV, OPENSTACK } from '../../../../../../../../../../common/constants';
import { SSH, VDDK } from '../../ConversionHostWizardConstants';

describe('conversion host wizard authentication step', () => {
  const store = generateStore({ form: formReducer }, {});

  const baseProps = {
    selectedProviderType: RHV,
    selectedTransformationMethod: null,
    verifyOpenstackCerts: null,
    store
  };

  const shallowDive = jsx => {
    const connectWrapper = shallow(jsx); // <Connect(Form(ConversionHostWizardAuthenticationStep))>
    const formWrapper = connectWrapper.dive(); // <Form(ConversionHostWizardAuthenticationStep)>
    const componentWrapper = formWrapper.dive(); // <ConversionHostWizardAuthenticationStep>
    return componentWrapper.dive(); // shallow-rendered ConversionHostWizardAuthenticationStep
  };

  it('renders the redux-form wrapper correctly', () => {
    const component = shallow(<ConversionHostWizardAuthenticationStep {...baseProps} />);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly in the initial state for RHV', () => {
    const component = shallowDive(<ConversionHostWizardAuthenticationStep {...baseProps} />);
    expect(component.find('Field[controlId="openstack-user-input"]')).toHaveLength(0);
    expect(component.find('TextFileField[controlId="vmware-ssh-key-input"]')).toHaveLength(0);
    expect(component.find('Field[controlId="vddk-library-path"]')).toHaveLength(0);
    expect(component.find('Field[controlId="verify-openstack-certs"]')).toHaveLength(0);
    expect(component.find('TextFileField[controlId="openstack-ca-certs-input"]')).toHaveLength(0);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly in the initial state for OSP', () => {
    const component = shallowDive(
      <ConversionHostWizardAuthenticationStep {...baseProps} selectedProviderType={OPENSTACK} />
    );
    expect(component.find('Field[controlId="openstack-user-input"]')).toHaveLength(1);
    expect(component.find('TextFileField[controlId="vmware-ssh-key-input"]')).toHaveLength(0);
    expect(component.find('Field[controlId="vddk-library-path"]')).toHaveLength(0);
    expect(component.find('Field[controlId="verify-openstack-certs"]')).toHaveLength(1);
    expect(component.find('TextFileField[controlId="openstack-ca-certs-input"]')).toHaveLength(0);
    expect(component).toMatchSnapshot();
  });

  it('renders the verify TLS certs switch correctly', () => {
    const component = shallowDive(
      <ConversionHostWizardAuthenticationStep {...baseProps} selectedProviderType={OPENSTACK} />
    );
    const onSwitchChange = jest.fn();
    const switchRenderProp = component.find('Field[controlId="verify-openstack-certs"]').props().children;
    const renderedSwitch = switchRenderProp({ input: { value: false, onChange: onSwitchChange } });
    expect(renderedSwitch).toMatchSnapshot();
    mount(renderedSwitch)
      .props()
      .onChange(null, true);
    expect(onSwitchChange).toHaveBeenCalledWith(true);
  });

  it('renders correctly with verify OSP TLS certs turned on', () => {
    const component = shallowDive(
      <ConversionHostWizardAuthenticationStep {...baseProps} selectedProviderType={OPENSTACK} verifyOpenstackCerts />
    );
    expect(component.find('TextFileField[controlId="openstack-ca-certs-input"]')).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with SSH transformation method selected', () => {
    const component = shallowDive(
      <ConversionHostWizardAuthenticationStep {...baseProps} selectedTransformationMethod={SSH} />
    );
    expect(component.find('TextFileField[controlId="vmware-ssh-key-input"]')).toHaveLength(1);
    expect(component.find('Field[controlId="vddk-library-path"]')).toHaveLength(0);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with VDDK transformation method selected', () => {
    const component = shallowDive(
      <ConversionHostWizardAuthenticationStep {...baseProps} selectedTransformationMethod={VDDK} />
    );
    expect(component.find('TextFileField[controlId="vmware-ssh-key-input"]')).toHaveLength(0);
    expect(component.find('Field[controlId="vddk-library-path"]')).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
