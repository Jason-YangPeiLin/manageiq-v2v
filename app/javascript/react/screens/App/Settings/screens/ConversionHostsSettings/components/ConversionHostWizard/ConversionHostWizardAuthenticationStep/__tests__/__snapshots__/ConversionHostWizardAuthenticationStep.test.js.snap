// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion host wizard authentication step renders correctly in the initial state for OSP 1`] = `
<Form
  bsClass="form"
  className="form-horizontal"
  componentClass="form"
  horizontal={false}
  inline={false}
>
  <TextFileField
    controlId="host-ssh-key-input"
    controlWidth={7}
    help="Upload your SSH key file or paste its contents below."
    hideBody={false}
    info="This is the private key file used to connect to the conversion host instance for the OpenStack User."
    label="Conversion Host SSH private key"
    labelWidth={4}
    name="conversionHostSshKey"
  />
  <Field
    component={[Function]}
    controlWidth={7}
    inline_label={true}
    label="Transformation method"
    labelWidth={4}
    name="transformationMethod"
    option_key="id"
    option_value="name"
    options={
      Array [
        Object {
          "id": "VDDK",
          "name": "VDDK",
        },
        Object {
          "id": "SSH",
          "name": "SSH",
        },
      ]
    }
    required={true}
    style={
      Object {
        "marginTop": 25,
      }
    }
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    controlId="verify-ca-certs"
    controlWidth={7}
    label="Verify TLS Certificates"
    labelWidth={4}
    name="verifyCaCerts"
    style={
      Object {
        "marginTop": 25,
      }
    }
  >
    <Component />
  </Field>
</Form>
`;

exports[`conversion host wizard authentication step renders correctly in the initial state for RHV 1`] = `
<Form
  bsClass="form"
  className="form-horizontal"
  componentClass="form"
  horizontal={false}
  inline={false}
>
  <TextFileField
    controlId="host-ssh-key-input"
    controlWidth={7}
    help="Upload your SSH key file or paste its contents below."
    hideBody={false}
    info="RHV-M deploys a common SSH public key on all hosts when configuring them. This allows commands and playbooks to be run from RHV-M. The associated private key is in the file /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M."
    label="Conversion Host SSH private key"
    labelWidth={4}
    name="conversionHostSshKey"
  />
  <Field
    component={[Function]}
    controlWidth={7}
    inline_label={true}
    label="Transformation method"
    labelWidth={4}
    name="transformationMethod"
    option_key="id"
    option_value="name"
    options={
      Array [
        Object {
          "id": "VDDK",
          "name": "VDDK",
        },
        Object {
          "id": "SSH",
          "name": "SSH",
        },
      ]
    }
    required={true}
    style={
      Object {
        "marginTop": 25,
      }
    }
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    controlId="verify-ca-certs"
    controlWidth={7}
    label="Verify TLS Certificates"
    labelWidth={4}
    name="verifyCaCerts"
    style={
      Object {
        "marginTop": 25,
      }
    }
  >
    <Component />
  </Field>
</Form>
`;

exports[`conversion host wizard authentication step renders correctly with SSH transformation method selected 1`] = `
<Form
  bsClass="form"
  className="form-horizontal"
  componentClass="form"
  horizontal={false}
  inline={false}
>
  <TextFileField
    controlId="host-ssh-key-input"
    controlWidth={7}
    help="Upload your SSH key file or paste its contents below."
    hideBody={false}
    info="RHV-M deploys a common SSH public key on all hosts when configuring them. This allows commands and playbooks to be run from RHV-M. The associated private key is in the file /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M."
    label="Conversion Host SSH private key"
    labelWidth={4}
    name="conversionHostSshKey"
  />
  <Field
    component={[Function]}
    controlWidth={7}
    inline_label={true}
    label="Transformation method"
    labelWidth={4}
    name="transformationMethod"
    option_key="id"
    option_value="name"
    options={
      Array [
        Object {
          "id": "VDDK",
          "name": "VDDK",
        },
        Object {
          "id": "SSH",
          "name": "SSH",
        },
      ]
    }
    required={true}
    style={
      Object {
        "marginTop": 25,
      }
    }
    validate={
      Array [
        [Function],
      ]
    }
  />
  <TextFileField
    controlId="vmware-ssh-key-input"
    controlWidth={7}
    help="Upload your SSH key file or paste its contents below."
    hideBody={false}
    label="VMware hypervisors SSH private key"
    labelWidth={4}
    name="vmwareSshKey"
  />
  <Field
    component={[Function]}
    controlId="verify-ca-certs"
    controlWidth={7}
    label="Verify TLS Certificates"
    labelWidth={4}
    name="verifyCaCerts"
    style={
      Object {
        "marginTop": 25,
      }
    }
  >
    <Component />
  </Field>
</Form>
`;

exports[`conversion host wizard authentication step renders correctly with VDDK transformation method selected 1`] = `
<Form
  bsClass="form"
  className="form-horizontal"
  componentClass="form"
  horizontal={false}
  inline={false}
>
  <TextFileField
    controlId="host-ssh-key-input"
    controlWidth={7}
    help="Upload your SSH key file or paste its contents below."
    hideBody={false}
    info="RHV-M deploys a common SSH public key on all hosts when configuring them. This allows commands and playbooks to be run from RHV-M. The associated private key is in the file /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M."
    label="Conversion Host SSH private key"
    labelWidth={4}
    name="conversionHostSshKey"
  />
  <Field
    component={[Function]}
    controlWidth={7}
    inline_label={true}
    label="Transformation method"
    labelWidth={4}
    name="transformationMethod"
    option_key="id"
    option_value="name"
    options={
      Array [
        Object {
          "id": "VDDK",
          "name": "VDDK",
        },
        Object {
          "id": "SSH",
          "name": "SSH",
        },
      ]
    }
    required={true}
    style={
      Object {
        "marginTop": 25,
      }
    }
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    controlId="vddk-library-path"
    controlWidth={7}
    info="This is the location where you downloaded and saved the VDDK package, as a URL accessible via HTTP. e.g. http://hostname/VMware-vix-disklib-version.x86_64.tar.gz"
    label="VDDK library path"
    labelWidth={4}
    name="vddkLibraryPath"
    required={true}
    type="text"
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    controlId="verify-ca-certs"
    controlWidth={7}
    label="Verify TLS Certificates"
    labelWidth={4}
    name="verifyCaCerts"
    style={
      Object {
        "marginTop": 25,
      }
    }
  >
    <Component />
  </Field>
</Form>
`;

exports[`conversion host wizard authentication step renders correctly with verify TLS certs turned on 1`] = `
<Form
  bsClass="form"
  className="form-horizontal"
  componentClass="form"
  horizontal={false}
  inline={false}
>
  <TextFileField
    controlId="host-ssh-key-input"
    controlWidth={7}
    help="Upload your SSH key file or paste its contents below."
    hideBody={false}
    info="This is the private key file used to connect to the conversion host instance for the OpenStack User."
    label="Conversion Host SSH private key"
    labelWidth={4}
    name="conversionHostSshKey"
  />
  <Field
    component={[Function]}
    controlWidth={7}
    inline_label={true}
    label="Transformation method"
    labelWidth={4}
    name="transformationMethod"
    option_key="id"
    option_value="name"
    options={
      Array [
        Object {
          "id": "VDDK",
          "name": "VDDK",
        },
        Object {
          "id": "SSH",
          "name": "SSH",
        },
      ]
    }
    required={true}
    style={
      Object {
        "marginTop": 25,
      }
    }
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    controlId="verify-ca-certs"
    controlWidth={7}
    label="Verify TLS Certificates"
    labelWidth={4}
    name="verifyCaCerts"
    style={
      Object {
        "marginTop": 25,
      }
    }
  >
    <Component />
  </Field>
  <TextFileField
    controlId="ca-certs-input"
    controlWidth={7}
    help="Upload your certificates file, in PEM format, or paste its contents below."
    hideBody={false}
    label="Trusted CA Certificates"
    labelWidth={4}
    name="caCerts"
  />
</Form>
`;

exports[`conversion host wizard authentication step renders the redux-form wrapper correctly 1`] = `
<Form
  bsClass="form"
  className="form-horizontal"
  componentClass="form"
  horizontal={false}
  inline={false}
>
  <TextFileField
    controlId="host-ssh-key-input"
    controlWidth={7}
    help="Upload your SSH key file or paste its contents below."
    hideBody={false}
    info="RHV-M deploys a common SSH public key on all hosts when configuring them. This allows commands and playbooks to be run from RHV-M. The associated private key is in the file /etc/pki/ovirt-engine/keys/engine_id_rsa on RHV-M."
    label="Conversion Host SSH private key"
    labelWidth={4}
    name="conversionHostSshKey"
  />
  <Field
    component={[Function]}
    controlWidth={7}
    inline_label={true}
    label="Transformation method"
    labelWidth={4}
    name="transformationMethod"
    option_key="id"
    option_value="name"
    options={
      Array [
        Object {
          "id": "VDDK",
          "name": "VDDK",
        },
        Object {
          "id": "SSH",
          "name": "SSH",
        },
      ]
    }
    required={true}
    style={
      Object {
        "marginTop": 25,
      }
    }
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    controlId="verify-ca-certs"
    controlWidth={7}
    label="Verify TLS Certificates"
    labelWidth={4}
    name="verifyCaCerts"
    style={
      Object {
        "marginTop": 25,
      }
    }
  >
    <Component />
  </Field>
</Form>
`;
